modules/utxo-ord/src/inscriptions.ts

```diff
@@ -57,8 +67,105 @@ export function createOutputScriptForInscription(
     output: compiledScript,
     depth: 0,
   };
-  const payment = payments.p2tr({ redeems: [redeem] }, { eccLib });
+
+  return payments.p2tr({ redeems: [redeem], redeemIndex: 0 }, { eccLib });
+}
+
+export type InscriptionRevealData = {
+  address: string;
+  revealSize: number;
+};
+
+/**
+ * @returns InscriptionRevealData
+ * @param createInscriptionParams & network
+ */
+export function createInscriptionRevealData(
+  createInscriptionParams: CreateInscriptionParams & { network: Network }
+): InscriptionRevealData {
+  const payment = createPaymentForInscription(createInscriptionParams);
+
+  const { output: commitOutput, controlBlock } = payment;
+  assert(commitOutput);
+  assert(controlBlock);
+  const commitAddress = address.fromOutputScript(commitOutput, createInscriptionParams.network);
+
+  // create fake dummy txn for size
+  const psbt = bitgo.createPsbtForNetwork({ network: createInscriptionParams.network as Network });
+  psbt.addInput({
+    hash: Buffer.alloc(32),
+    index: 0,
+    witnessUtxo: { script: commitOutput, value: BigInt(10239812038) },
+    tapLeafScript: [
+      {
+        controlBlock,
+        script: commitOutput,
```

Comment:
script here should be the inscription reveal script, which I think ends up in `payment.redeem.output`.